### PR TITLE
Add the option to use existing secrets for DB, imagePullSecrets and license file, fixes #61

### LINKED ISSUE #61: Support pre-existing kubernetes secrets
Currently, two option exist for referencing the License-File and DB-/Admin-credentials:

1. Via the secrets-store.csi.x-k8s.io/v1 API.
2. As a Kubernetes secret, directly created by the Helm-Chart.

It would be helpful to introduce a third option where a pre-existing Kubernetes secret can be referenced. This would be beneficial when other Azure Key-Vault synchronization mechanisms are in use, like akv2k8s (https://akv2k8s.io/).

### PR DESCRIPTION
**Pull request by mpsOxygen**
This PR adds the following functionality (by default they are disabled):

- the ability to use a preexisting imagePullSecrets thru these values
```
  imagePullSecrets:
    # -- The base64 encoded dockerconfigjson
    secret: ""
    # -- Use an existing imagePullSecret created by a secrets operator
    existingSecret:
      # -- By default the existing secret is disabled, set this to true in order to provide an existing imagePullSecret
      enabled: false
      # -- The existing imagePullSecret name
      name: ""
```

- the ability to use a preexisting secret for database connection details thru these values:
```
    additionalEnv:
      null
      # -- Uncomment these when using secret.existingDbSecret.enabled = true
      # - name: DB_NAME
      #   valueFrom:
      #     secretKeyRef:
      #       name: secret-name
      #       key: secret-key
      # - name: DB_HOST
      #   valueFrom:
      #     secretKeyRef:
      #       name: secret-name
      #       key: secret-key
      # - name: DB_USER
      #   valueFrom:
      #     secretKeyRef:
      #       name: secret-name
      #       key: secret-key
      # - name: DB_PASSWORD
      #   valueFrom:
      #     secretKeyRef:
      #       name: secret-name
      #       key: secret-key
...
  secret:
    dbSecret:
      enabled: false # Enable to apply database-secret.yaml which allows you to specify db credentials
    # -- Use an existing secret for the database variables: DB_NAME, DB_HOST, DB_USER, DB_PASSWORD
    existingDbSecret:
      # -- Enabled the use of the already created secret
      enabled: true
```

- the ability to use a preexisting secret for the license file:
```
  license:
    name: nexus-repo-license.lic
    # -- Use an already created secret with the license contents in base64, it takes the name from the above field -> name: nexus-repo-license.lic and the mount path from below: licenseSecret.mountPath
    existingSecret: false
```

Also there are a couple of formatting fixes mostly done automatically by the IDE.

This has been tested in my environment and works as expected, also I have tested that it does not impact existing functionality.

### CODE REVIEW

**Comment by bobotimi** [comment]
@mpsOxygen Please rebase your PR to pull in the latest changes on main and resolve the conflicts

**Review by bobotimi — APPROVED** [review]

File: nxrm-ha/Chart.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 72.0.0
+version: 72.0.1
```
[the remark this answers is not in the document]
Comment (mpsOxygen) [reply nxrm-ha/Chart.yaml]: Done

**Comment by bobotimi** [comment]
@mpsOxygen This has now been addressed in https://github.com/sonatype/nxrm3-ha-repository/pull/112, you may close your PR

**Comment by bobotimi** [comment]
@mpsOxygen This has now been addressed in https://github.com/sonatype/nxrm3-ha-repository/pull/112, closing your PR

**Pull request closed without merging**